scaffoldmaker/meshtypes/meshtype_3d_ellipticcylinder1.py

```diff
@@ -0,0 +1,476 @@
+"""
+Generates a 3-D elliptic cylinder model.
+"""
+
+from __future__ import division
+
+import math
+
+from scaffoldmaker.meshtypes.scaffold_base import Scaffold_base
+from scaffoldmaker.utils.meshrefinement import MeshRefinement
+from scaffoldmaker.utils.eftfactory_bicubichermitelinear import eftfactory_bicubichermitelinear
+from scaffoldmaker.utils.eftfactory_tricubichermite import eftfactory_tricubichermite
+from scaffoldmaker.utils.zinc_utils import *
+from scaffoldmaker.utils.geometry import *
+from scaffoldmaker.utils.interpolation import *
+from scaffoldmaker.utils.vector import *
+from scaffoldmaker.utils.transformation import *
+
+from opencmiss.zinc.element import Element, Elementbasis
+from opencmiss.zinc.field import Field
+from opencmiss.zinc.node import Node
+
+
+class MeshType_3d_ellipticcylinder1(Scaffold_base):
+    """
+    Generates a 3-D vertebral mesh
+    """
+
+    @staticmethod
+    def getName():
+        return '3D Elliptic Cylinder 1'
+
+    @staticmethod
+    def getDefaultOptions(parameterSetName='Default'):
+        return {'Number of elements around': 4, 'Number of elements up': 3, 'Number of elements through wall': 2,
+                'Major diameter': 1.2, 'Minor diameter': 0.7, 'Height': 2.0, 'Thickness ratio': 0.5,
```

Comment:
'Thickness ratio' is too vague. Qualify with other words e.g. Longitudinal & what it's a ratio of end to middle?